packages/ubuntu_wsl_setup/lib/services/tcp_helpers.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:async';
+import 'dart:io';
+
+// This is inherently a race condition on its own, because in between finding an
+// available port and using it, the system may have binded other socket.
+FutureOr<RawServerSocket?> _tryBindUnsafe({
+  required int lower,
+  required int higher,
+}) async {
+  final socket = await RawServerSocket.bind(InternetAddress.loopbackIPv4, 0);
+  final port = socket.port;
+  if (port > lower && port < higher) {
```

Comment:
I want to have some randomness in the port selection, thus I deferred to the OS to give me a port in the ephemeral range, but since the ranges are not equal between Windows and Linux I made those values as customization points for callers. My intended usage will be using Windows lower bound and Linux upper bound (the intersection of both ranges). In fact, the stricter the range, more prone to fail.
Another approach would be using random number generation to always generate ports inside that range and them attempt to bind. Actually after some experimentation I'm observing a monotonically growing trend on the Windows side, always returning higher and higher ports. Since Windows may generate 4536 ports beyond Linux's upper limit the current algorithm could easily experience a premature (and even not real) port exaustion. Even because every port attempted stays blocked after close for a certain amount of time (OS security feature).
I'll go with the random number generation.